brainiak/reprsimil/brsa.py

```diff
@@ -0,0 +1,1118 @@
+        (e.g., calculating the similarity matrix of responses to each event),
+        you might want to start with specifying a lower rank and use metrics
+        such as AIC or BIC to decide the optimal rank.
+    GP_space: boolean, default: False
+        Whether to impose a Gaussion Process (GP) prior on the log(pseudo-SNR).
+        If true, the GP has a kernel defined over spatial coordinate.
+        This is experimental and slow. I find that when SNR is genrally low,
+        smoothness can be overestimated. But I think this is better than
+        not imposing any regularization.
+    GP_inten: boolean, defualt: False
+        Whether to include a kernel definved over the intensity of image.
+        GP_space should be True as well if you want to use this,
+        because the smoothness should be primarily in space.
+        Smoothness in intensity is just complementary.
+    verbose : boolean, default: False
+        Verbose mode flag.
+    pad_DC: boolean, default: False
+        A column of all 1's will be padded to end of the design matrix
+        to account for residual baseline component in the signal.
+        We recommend removing DC component in your data but still
+        set this as True. If you include a baseline column yourself,
+        then you should check this as False.
+        In future version, we will include a seperate input
+        argument for all regressors you are not interested in,
+        such as DC component and motion parameters.
+
+    Attributes
+    ----------
+    U_ : The shared covariance matrix, shape=[condition,condition].
+    L_ : The Cholesky factor of the shared covariance matrix
+        (lower-triangular matrix), shape=[condition,condition].
+    C_: the correlation matrix derived from the shared covariance matrix,
+        shape=[condition,condition]
+    nSNR_ : array, shape=[voxels,]
+        The pseuso-SNR of all voxels.
+        They are normalized such that the geometric mean is 1
+    sigma_ : array, shape=[voxels,]
+        The estimated standard deviation of the noise in each voxel
+        Assuming AR(1) model, this means the standard deviation
+        of the refreshing noise.
+    rho_ : array, shape=[voxels,]
+        The estimated autoregressive coefficient of each voxel
+    lGPspace_ : number, only if GP_space or GP_inten is True
+        the length scale of Gaussian Process prior of log(SNR)
+    bGP_ : number, only if GP_space or GP_inten is True.
+        the standard deviation of the GP prior
+    lGPinten_: number, only if GP_inten is True
+        the length scale in fMRI intensity of the GP prior of log(SNR)
+
+    Notes
+    -----
+    The current version assumes noise is independent across voxels.
+    Real data typically has spatial correlation in noise.
+    This assumption might still introduce some bias in the result.
+    Spatial correlation will be included in a future version.
+    """
+
+    def __init__(
```

Comment:
Thanks Mike! I will add them! Those are parameters added later and I forgot
to add documentation.

Mingbo Cai


On Thu, Sep 8, 2016 at 1:39 PM, Michael Shvartsman <notifications@github.com
> wrote:

> In brainiak/reprsimil/brsa.py
> <https://github.com/IntelPNI/brainiak/pull/98#discussion_r78053771>:
>
> > +    lGPspace_ : number, only if GP_space or GP_inten is True
> > +        the length scale of Gaussian Process prior of log(SNR)
> > +    bGP_ : number, only if GP_space or GP_inten is True.
> > +        the standard deviation of the GP prior
> > +    lGPinten_: number, only if GP_inten is True
> > +        the length scale in fMRI intensity of the GP prior of log(SNR)
> > +
> > +    Notes
> > +    -----
> > +    The current version assumes noise is independent across voxels.
> > +    Real data typically has spatial correlation in noise.
> > +    This assumption might still introduce some bias in the result.
> > +    Spatial correlation will be included in a future version.
> > +    """
> > +
> > +    def __init__(
>
> Looks like there are some parameters here (intern_smooth_range,
> space_smooth_range, etc) that are not documented up in the docstring. If I
> understand the standard correctly, all exposed parameters (e.g. arguments
> to *init_*) should be documented.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/IntelPNI/brainiak/pull/98/files/1f5775a6846235c51d7c65be2e22eae6cdec5444#r78053771>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHZ3TXs1FwywlNuQwu1-cgF-TbXEiMklks5qoEg_gaJpZM4J3eAD>
> .
>